### PR TITLE
Updating RepoToolset to force public signing on non-windows

### PR DESCRIPTION
**Pull request by tannergooding**
FYI. @jaredpar, @tmat

### CODE REVIEW

File: sdks/RepoToolset/tools/StrongName.targets [hunk 1]
```diff
@@ -50,6 +50,11 @@
     </When>
   </Choose>
 
+  <!-- Full signing is not currently supported on non-Windows -->
```
Comment (tmat) [inline sdks/RepoToolset/tools/StrongName.targets]: I believe the compilers already support it. Shall we move to the newer compilers version instead?

Comment (tannergooding) [reply sdks/RepoToolset/tools/StrongName.targets]: Some reps aren't using the tool set compilers and are using a particular cli build instead.

Comment (tannergooding) [reply sdks/RepoToolset/tools/StrongName.targets]: Not in all cases. For something like release/2.0.0, we are explicitly building with a 2.0 build of the cli.

Comment (tmat) [reply sdks/RepoToolset/tools/StrongName.targets]: Then I'd say the repo should set this. We want to use the latest compilers and features by default.

Comment (jaredpar) [reply sdks/RepoToolset/tools/StrongName.targets]: I agree.

IMHO: The lag of toolset adoption means that either RepoToolset or individual repos need to tweak certain settings. That will be true for this change and others in the future. I would rather make it more painful for repos who are not adopting completely compatible toolsets than add too much complexity to the central tools.

**Pull request merged**